flink-ai-flow/ai_flow/api/ops.py

```diff
@@ -0,0 +1,582 @@
+
+
+def write_dataset(input_data: Channel,
+                  dataset_info: Union[DatasetMeta, Text, int],
+                  executor=None,
+                  **kwargs
+                  ) -> NoneChannel:
+    """
+    Write dataset to dataset operator. It can write dataset to external system.
+
+    :param input_data: Channel from the specific operator which generates data.
+    :param dataset_info: Information about the dataset which will be read. Its type can be DataSetMeta
+                         of py:class:`ai_flow.meta.dataset_meta.DatasetMeta` or Text or int. The dataset_info
+                         means name in he metadata service when its type is Text and it means id when its type is int.
+                         The ai flow will get the dataset from metadata service by name or id.
+    :param executor: The python user defined function in write dataset operator. User can write their own logic here.
+    :return: NoneChannel.
+    """
+    if isinstance(dataset_info, DatasetMeta):
+        dataset_meta = dataset_info
+    elif isinstance(dataset_info, Text):
+        dataset_meta = get_ai_flow_client().get_dataset_by_name(dataset_info)
+    else:
+        dataset_meta = get_ai_flow_client().get_dataset_by_id(dataset_info)
+
+    return user_define_operation(input_data=input_data,
+                                 dataset_info=dataset_meta,
+                                 executor=executor,
+                                 output_num=0,
+                                 operation_type='write_dataset',
+                                 **kwargs)
+
+
+def transform(input_data: Union[Channel, List[Channel]],
+              executor,
+              output_num=1,
+              name: Text = None,
+              **kwargs) -> Union[Channel, Tuple[Channel]]:
+    """
+    Transformer operator. Transform the dataset so that the original dataset can be used for trainer or other operators
+    after feature engineering, data cleaning or some other data transformation.
+
+    :param input_data: List of input data. It contains multiple channels from the operators which generate data.
+    :param executor: The user defined function in transform operator. User can write their own logic here.
+    :param output_num: The output number of the operator. The default value is 1.
+    :param name: Name of the transform operator.
+    :return: Channel or Tuple[Channel]. It returns Channel When the output_num is 1 and returns Tuple[Channel] when
+             the output_num is bigger than 1, and the len(Tuple(Channel)) is output_num.
+    """
+    return user_define_operation(input_data=input_data,
+                                 name=name,
+                                 executor=executor,
+                                 output_num=output_num,
+                                 operation_type='transform',
+                                 **kwargs)
+
+
+def train(input_data: Union[Channel, List[Channel]],
```

Comment:
Would it be better to have a more meaningful name for the parameters? e.g. `training_samples`. Similarly, for the `Executor`, it could be `training_executor`. This also applies to the other ops.